### PR TITLE
halloy: 2024.08 -> 2024.11

### PR DESCRIPTION
**Pull request by shombando**
## Description of changes

Updated halloy version to [2024.11](https://github.com/squidowl/halloy/releases/tag/2024.11) (most recent release). Used the Cargo.lock from the tagged release and added a new build dependency alsa-lib.

Diff: https://github.com/squidowl/halloy/compare/2024.8...2024.11
Changelog: https://github.com/squidowl/halloy/releases/tag/2024.11

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

---
This is my first Nix related PR, I would appreciate any feedback or things I missed from the contributing docs. Thank you!

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 341510` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>halloy</li>
  </ul>
</details>

**Review by fabaff — COMMENTED** [review]
Please update the commit message to match the guidelines.

**Comment by shombando** [comment]
@fabaff thanks for the feedback, I've updated the commit message.

File: pkgs/by-name/ha/halloy/package.nix [hunk 1]
```diff
@@ -43,6 +44,7 @@ rustPlatform.buildRustPackage rec {
 
   buildInputs =
     [
+      alsa-lib.dev
```
Comment (Scrumplex) [inline pkgs/by-name/ha/halloy/package.nix:47]: .dev should be used by default here. See: https://github.com/NixOS/nixpkgs/blob/c59414d3bb6630300418fac44e201ca638863aaf/pkgs/stdenv/generic/make-derivation.nix#L304-L331

```suggestion
      alsa-lib
```

**Review by Scrumplex — COMMENTED** [review]
I couldn't get the app running on my machine. It would segfault immediately during launch :thinking: 

Edit: same issue with the previous (currently packaged) version, so this is not a blocker!

**Comment by shombando** [comment]
@Scrumplex that's odd, both the existing nixpkgs and this version work for me (in nix-shell and as installed). I wonder if the application is looking for a config file that doesn't exist, I installed mine via home-manager so both versions have that available at startup.
I'm happy to do some testing if you have any ideas.

**Comment by Scrumplex** [comment]
> I'm happy to do some testing if you have any ideas.

Looking at the stack trace reveals that it is likely some issue with my compositor. I am running Sway 1.9 from Nixpkgs nixos-unstable.

```
                Stack trace of thread 906139:
                #0  0x00007c90974ba2a0 wl_proxy_get_version (libwayland-client.so.0 + 0x82a0)
                #1  0x00007c908d160410 wsi_wl_surface_create_swapchain (libvulkan_radeon.so + 0x160410)
                #2  0x00007c908d154358 wsi_CreateSwapchainKHR (libvulkan_radeon.so + 0x154358)
                #3  0x00007c9096ab844a terminator_CreateSwapchainKHR (libvulkan.so.1 + 0x4244a)
                #4  0x00007c9084836d0d _ZL26overlay_CreateSwapchainKHRP10VkDevice_TPK24VkSwapchainCreateInfoKHRPK21VkAllocationCallbacksPP16VkSwapchainKHR_T (libMangoHud.so + 0x36d0d)
                #5  0x0000643720f5385e _ZN8wgpu_hal6vulkan8instance102_$LT$impl$u20$wgpu_hal..Surface$LT$wgpu_hal..vulkan..Api$GT$$u20$for$u20$wgpu_hal..vulkan..Surface$GT$9configure17h1dfeb0b7b34e492fE (halloy + 0xbd085e)
                #6  0x0000643720e172e3 _ZN9wgpu_core6device6global52_$LT$impl$u20$wgpu_core..global..Global$LT$G$GT$$GT$17surface_configure17h9fceb14981732d72E (halloy + 0xa942e3)
                #7  0x0000643720ee97b9 _ZN84_$LT$wgpu..backend..wgpu_core..ContextWgpuCore$u20$as$u20$wgpu..context..Context$GT$17surface_configure17hddd0c9b55a0524e1E (halloy + 0xb667b9)
                #8  0x0000643720ef7503 _ZN47_$LT$T$u20$as$u20$wgpu..context..DynContext$GT$17surface_configure17hef1dc581fa886fe1E (halloy + 0xb74503)
                #9  0x0000643720d7b561 _ZN4wgpu7Surface9configure17h71c626802b087264E (halloy + 0x9f8561)
                #10 0x0000643720ad3d1d _ZN99_$LT$iced_wgpu..window..compositor..Compositor$u20$as$u20$iced_graphics..compositor..Compositor$GT$17configure_surface17h8c26c9ab0c2f95f0E (halloy + 0x750d1d)
                #11 0x0000643720677248 _ZN99_$LT$iced_wgpu..window..compositor..Compositor$u20$as$u20$iced_graphics..compositor..Compositor$GT$14create_surface17h13929c522f18b510E (halloy + 0x2f4248)
                #12 0x0000643720693a1b _ZN10iced_winit11application12run_instance28_$u7b$$u7b$closure$u7d$$u7d$17h396fbf415e517f2aE.llvm.12253500630602256029 (halloy + 0x310a1b)
                #13 0x0000643720698bda _ZN10iced_winit11application3run29Runner$LT$Message$C$F$C$C$GT$13process_event17h4bf20cf40a389a43E.llvm.12253500630602256029 (halloy + 0x315bda)
                #14 0x000064372066d4d5 _ZN4core3ops8function5impls79_$LT$impl$u20$core..ops..function..FnMut$LT$A$GT$$u20$for$u20$$RF$mut$u20$F$GT$8call_mut17h883c4b940620fa1cE (halloy + 0x2ea4d5)
                #15 0x00006437205db4d8 _ZN5winit13platform_impl8platform7wayland10event_loop18EventLoop$LT$T$GT$11pump_events17h98d4216319812e33E (halloy + 0x2584d8)
                #16 0x00006437205cb72b _ZN5winit13platform_impl8platform18EventLoop$LT$T$GT$3run17ha475fd458da29673E (halloy + 0x24872b)
                #17 0x00006437206c1ae8 _ZN6halloy4main17hde7870d69e39e15cE (halloy + 0x33eae8)
                #18 0x00006437206ea746 _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd19b368670afaff6E (halloy + 0x367746)
                #19 0x000064372069e9d5 _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17he446a48f64c75213E (halloy + 0x31b9d5)
                #20 0x0000643721622977 _ZN3std9panicking3try17h313170bb1d1d6e4bE (halloy + 0x129f977)
                #21 0x0000643721631c1d _ZN3std2rt19lang_start_internal17h6cc7d681663e7b4eE (halloy + 0x12aec1d)
                #22 0x00006437206c5b55 main (halloy + 0x342b55)
                #23 0x00007c9096d4d14e __libc_start_call_main (libc.so.6 + 0x2a14e)
                #24 0x00007c9096d4d209 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2a209)
                #25 0x00006437205579a5 _start (halloy + 0x1d49a5)
```

**Comment by Scrumplex** [comment]
I found the issue right after posting my comment. It's my globally enabled MangoHud

**Comment by Scrumplex** [comment]
Apart from my suggestion above this LGTM, builds and runs (with MangoHud disabled)!

File: pkgs/by-name/ha/halloy/package.nix (hunk 1 above)
Comment (shombando) [reply pkgs/by-name/ha/halloy/package.nix]: @Scrumplex thank you, I have made the change and ammended the commit.

**Review by Scrumplex — APPROVED** [review]

**Comment by shombando** [comment]
@fabaff hello, I'd like your insight. Halloy has released 2024.12 in the meantime. I think I'd still like to go through with this pull request and perhaps submit another one for the new version. I'd like to test 2024.12 locally for a bit first. Does that match your expectations?

**Comment by fabaff** [comment]
@shombando, do it the way that works best for you.

**Review by fabaff — APPROVED** [review]
Change looks good to me.

**Pull request merged**